services/src/main/java/org/keycloak/authentication/authenticators/client/KubernetesJWTClientValidator.java

```diff
@@ -0,0 +1,69 @@
+package org.keycloak.authentication.authenticators.client;
+
+import org.keycloak.OAuth2Constants;
+import org.keycloak.authentication.AuthenticationFlowError;
+import org.keycloak.authentication.ClientAuthenticationFlowContext;
+import org.keycloak.protocol.oidc.OIDCAdvancedConfigWrapper;
+import org.keycloak.services.ServicesLogger;
+import org.keycloak.utils.StringUtil;
+
+/**
+ * Validator for Kubernetes Signed JWTs using the Kubernetes JWKS endpoint.
+ *
+ * @author <a href="mailto:sebastian.laskawiec@defenseunicorns.com></a>
+ */
+public class KubernetesJWTClientValidator extends JWTClientValidator {
+
+    public KubernetesJWTClientValidator(ClientAuthenticationFlowContext context, String clientAuthenticatorProviderId) {
+        super(context, clientAuthenticatorProviderId);
+    }
+
+    /**
+     * Kubernetes Signed JWTs shouldn't be checked against reuse. They are typically mounted into the Pod and depending
+     * on the workload application can be cached and reused or not.
+     * @return
+     */
+    @Override
+    protected boolean isJwtIdRequired() {
+        return false;
```

Comment:
This is not enough. We need to explicitly disable duplicate token detection. Otherwise apps with mounted tokens from Kube won't be able to use it.